doc/userguide/rules/datasets.rst

```diff
@@ -0,0 +1,152 @@
+Datasets
+========
+
+Using the ``dataset`` and ``datarep`` keyword it is possible to match on
+large amounts of data against any sticky buffer.
+
+For example, to match against a dns black list called ``dns-bl``::
+
+    dns.query; dataset:isset,dns-bl;
+
+These keywords are aware of transforms. So to look up a DNS query again
+a MD5 black list::
+
+    dns.query; to_md5; dataset:isset,dns-bl;
+
+Global config (optional)
+------------------------
+
+Datasets can optionally be defined in the main config. Sets can also be
+declared from the rule syntax.
+
+Example of sets for tracking unique values::
+
+    datasets:
+     - ua-seen:
+       type: string
+       state: ua-seen.lst
+     - dns-sha256-seen:
+       type: sha256
+       state: dns-sha256-seen.lst
+
+Rules to go with the above::
+
+    http.user_agent; dataset:set,ua-seen;
+    dns.query; to_sha256; dataset:set,dns-sha256-seen;
+
+
+Rule keywords
+-------------
+
+dataset
+~~~~~~~
+
+Datasets are binary: something is in the set or it's not.
+
+Syntax::
+
+    dataset:<cmd>,<name>,<options>;
+
+    dataset:<set|isset|isnotset>,<name> \
+        [, type <string|md5|sha256>, save <file name>, load <file name>, state <file name>];
+
+type <type>
+  the data type: string, md5, sha256
+save <file name>
+  file name for saving the in memory data when Suricata exits
```

Comment:
`the in` is probably in reverse order

Reply:
Maybe I should have written 'in-memory'.